test/Conversion/CombToLLVM/comb-to-llvm.mlir

```diff
@@ -0,0 +1,215 @@
+  return %0, %1, %2 : i32, i32, i32
+}
+
+// CHECK-LABEL: llvm.func @test_comb_icmp
+func.func @test_comb_icmp(%arg0: i32, %arg1: i32) -> (i1, i1, i1, i1) {
+  // Test comparison operations (converted via Comb->Arith->LLVM)
+  // CHECK: llvm.icmp "eq"
+  // CHECK: llvm.icmp "ne"
+  // CHECK: llvm.icmp "slt"
+  // CHECK: llvm.icmp "ult"
+  // CHECK: llvm.return
+  %0 = comb.icmp eq %arg0, %arg1 : i32
+  %1 = comb.icmp ne %arg0, %arg1 : i32
+  %2 = comb.icmp slt %arg0, %arg1 : i32
+  %3 = comb.icmp ult %arg0, %arg1 : i32
+  return %0, %1, %2, %3 : i1, i1, i1, i1
+}
+
+// CHECK-LABEL: llvm.func @test_comb_mux
+func.func @test_comb_mux(%cond: i1, %arg0: i32, %arg1: i32) -> i32 {
+  // Test mux operation (converted via Comb->Arith->LLVM)
+  // CHECK: llvm.select
+  // CHECK: llvm.return
+  %0 = comb.mux %cond, %arg0, %arg1 : i32
+  return %0 : i32
+}
+
+// CHECK-LABEL: llvm.func @test_comb_replicate
+func.func @test_comb_replicate(%arg0: i4) -> i16 {
+  // Test replicate operation (converted via Comb->Arith->LLVM)
+  // CHECK: llvm.zext
+  // CHECK: llvm.or
+  // CHECK: llvm.return
+  %0 = comb.replicate %arg0 : (i4) -> i16
+  return %0 : i16
+}
+
+// CHECK-LABEL: llvm.func @test_variadic_operations
+func.func @test_variadic_operations(%arg0: i32, %arg1: i32, %arg2: i32) -> (i32, i32, i32) {
+  // Test variadic operations (converted via Comb->Arith->LLVM)
+  // CHECK: llvm.add
+  // CHECK: llvm.add
+  // CHECK: llvm.mul
+  // CHECK: llvm.mul
+  // CHECK: llvm.and
+  // CHECK: llvm.and
+  // CHECK: llvm.return
+  %0 = comb.add %arg0, %arg1, %arg2 : i32
+  %1 = comb.mul %arg0, %arg1, %arg2 : i32
+  %2 = comb.and %arg0, %arg1, %arg2 : i32
+  return %0, %1, %2 : i32, i32, i32
+}
+
+// CHECK-LABEL: llvm.func @test_control_flow
+func.func @test_control_flow(%cond: i1, %arg0: i32) -> i32 {
+  // Test that control flow is also converted (SCF->ControlFlow->LLVM)
+  // CHECK: llvm.cond_br
+  // CHECK: llvm.br
+  // CHECK: llvm.return
+  %result = scf.if %cond -> i32 {
+    %0 = comb.add %arg0, %arg0 : i32
+    scf.yield %0 : i32
+  } else {
+    %1 = comb.mul %arg0, %arg0 : i32
+    scf.yield %1 : i32
+  }
+  return %result : i32
+}
+
+// -----
+
+// Test HW module and SV operations (should NOT be converted - only func.func operations are converted)
+hw.module @test_hw_sv_module(in %scan_data : i9) {
+  %0 = comb.extract %scan_data from 0 : (i9) -> i8
+  %top_reg_decoded = sv.wire : !hw.inout<i8>
+  sv.assign %top_reg_decoded, %0 : i8
+  %1 = comb.extract %scan_data from 8 : (i9) -> i1
+  %child.child_reg_decoded = sv.wire : !hw.inout<i1>
+  sv.assign %child.child_reg_decoded, %1 : i1
+  %2 = comb.extract %scan_data from 8 : (i9) -> i1
+  hw.instance "child_decoder" @test_child_module(scan_data: %2: i1) -> ()
+  hw.output
+}
+
+hw.module @test_child_module(in %scan_data : i1) {
+  %0 = comb.extract %scan_data from 0 : (i1) -> i1
+  %SimpleChild.child_reg_decoded = sv.wire : !hw.inout<i1>
+  sv.assign %SimpleChild.child_reg_decoded, %0 : i1
+  hw.output
+}
```

Comment:
Super-Nit: we can probably reduce that test a bit

Reply:
Thanks for noticing, this is not required. Reduced it.